### PR TITLE
feat: simple login and signup

### PR DESCRIPTION
**Pull request by stephenhungg**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- login page, email/password authentication with password toggle
- signup page, multi field registration with first name, last name, phone number, email, and password

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

signup
login

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

/src/app/auth/login/page.tsx
/src/app/auth/signup/page.tsx

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Better design

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

chat gpt

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

Yuhhhhhh

CC: @rachaelch3n

### CODE REVIEW

File: src/app/auth/signup/page.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+        <Button
+          text="Continue"
+          disabled={
+            !email || !password || !first_name || !last_name || !phone_number
+          }
+          onPress={handleSignUp}
+        />
+        <Text style={styles.linkText}>
+          Need an account?{' '}
```
Comment (torinnayak1) [inline src/app/auth/signup/page.tsx]: This should be "Have an Account?"

**Review by torinnayak1 — APPROVED** [review]
Holy LLM final boss

Fix the text for the "Need an account?" on the sign up page and it's good

File: src/app/auth/signup/page.tsx (hunk 1 above)
Comment (stephenhungg) [reply src/app/auth/signup/page.tsx]: Type type type

File: src/app/auth/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          <Text style={styles.link} onPress={handleSignUp}>
+            Sign up
+          </Text>
+        </Text>
+      </View>
+    </SafeAreaView>
+  );
+}
+
+const styles = StyleSheet.create({
```
Comment (kevin3656) [inline src/app/auth/login/page.tsx]: since these stylings are reused in the signup page maybe the can be moved to a separate styling file and imported from there ? Speaking from prev web dev experience not sure if thats how it works in react native though lol.

**Review by kevin3656 — APPROVED** [review]
Looks good !

File: src/app/auth/signup/page.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+import {
+  Alert,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { useNavigation } from '@react-navigation/native';
```
Comment (rachaelch3n) [inline src/app/auth/signup/page.tsx]: we are going to be using expo router instead of react navigation

File: api/rails/client.ts
Comment (rachaelch3n) [inline api/rails/client.ts:1]: thanks for renaming

File: api/config.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
+// API Configuration for Rails Backend
+const BASE_URL =
+  process.env.EXPO_PUBLIC_API_BASE_URL || 'http://10.2.193.251:3000';
```
Comment (rachaelch3n) [inline api/config.ts]: looks good for local testing! just noting that the fallback ip should be replaced with your own machine's ip when testing locally, and later updated to the deployed api url before production

File: env.example [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Rails API Configuration
+EXPO_PUBLIC_API_BASE_URL=http://localhost:3000
```
Comment (rachaelch3n) [inline env.example:2]: also noting here that the base url will be updated for deployment

File: src/app/auth/login/page.tsx (hunk 2 above)
Comment (rachaelch3n) [reply src/app/auth/login/page.tsx]: yes! can we make a new file like src/styles/authStyles.ts and add all the shared styling code there? then just import and replace all the styles with authStyles (and delete the style sheets).

styling is also subject to change right now, so this file might change in the future

File: src/app/auth/signup/page.tsx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  logo: {
+    fontSize: 24,
+    fontWeight: 'bold',
+    textAlign: 'center',
+    marginBottom: 10,
+  },
+});
```
Comment (rachaelch3n) [inline src/app/auth/signup/page.tsx]: just flagging this to put into the new file

File: src/app/auth/login/page.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        error instanceof Error
+          ? error.message
+          : 'Invalid email or password. Please try again.',
+      );
+    }
+  };
+
+  const handleSignUp = () => {
+    router.push('/auth/signup/page');
+  };
```
Comment (rachaelch3n) [inline src/app/auth/login/page.tsx]: maybe rename the function to something like navigateToSignUp so its clearer that the function reroutes, instead of handling signup form submission

File: src/app/auth/signup/page.tsx [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+      Alert.alert(
+        'Signup Failed',
+        error instanceof Error
+          ? error.message
+          : 'Failed to create account. Please try again.',
+      );
+    }
+  };
+
+  const handleSignIn = () => {
```
Comment (rachaelch3n) [inline src/app/auth/signup/page.tsx]: can we also rename this to navigateToSIgnIn

Comment (stephenhungg) [reply src/app/auth/signup/page.tsx]: Yes

File: src/app/auth/login/page.tsx (hunk 7 above)
Comment (stephenhungg) [reply src/app/auth/login/page.tsx]: Ok

File: api/config.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+export const API_ENDPOINTS = {
+  LOGIN: '/api/login',
+  DRIVERS: '/api/drivers',
+} as const;
+
+// Types for driver auth
+export interface DriverSignupData {
+  email: string;
+  password: string;
+  password_confirmation: string;
```
Comment (rachaelch3n) [inline api/config.ts:17]: noting that there's no visible input for password_confirmation right now, but will add it later

File: src/app/auth/signup/page.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+      Alert.alert(
+        'Signup Failed',
+        error instanceof Error
+          ? error.message
+          : 'Failed to create account. Please try again.',
+      );
+    }
+  };
+
+  const navigateToLogin = () => {
```
Comment (kylezryr) [inline src/app/auth/signup/page.tsx]: since this function is only one line it might be better to put the router.push call directly in the onclick of the button

File: src/app/_layout.tsx [hunk 11]
```diff
@@ -1,26 +1,11 @@
...
-    <SafeAreaProvider>
-      <Stack.Navigator screenOptions={{ headerShown: false }}>
-        <Stack.Screen name="Index" component={IndexScreen} />
-      </Stack.Navigator>
-    </SafeAreaProvider>
+    <Stack screenOptions={{ headerShown: false }}>
+      <Stack.Screen name="index" options={{ headerShown: false }} />
+      <Stack.Screen name="auth/login/page" />
+      <Stack.Screen name="auth/signup/page" />
+    </Stack>
```
Comment (kylezryr) [inline src/app/_layout.tsx:9]: i think you almost have the right idea for expo router stack navigation, but some tweaks need to be made:

1. add a `_layout.tsx` file to each of the new screen folders  (/login and /signup), and rename `page.tsx` to `index.tsx`
2. in the `_layout.tsx` file, you can past this as the content:
```
import { Stack } from 'expo-router';

function StackLayout() {
  return (
    <Stack>
      <Stack.Screen name="index" options={{ headerShown: false }} />
    </Stack>
  );
}

export default StackLayout;
```
this basically exports the `index.tsx` file as the screen itself, so we can go back to `app/_layout.tsx` and rename the Stacks to `name="auth/login"` and `name="auth/signup"` respectively. 
3. this allows us to use router to route to `auth/signup` or `auth/login` instead of adding the `/page` at the end, which is kinda bad practice

lmk if this doesn't make sense, if you need more examples take a look at [Girls Write Now's codebase](https://github.com/calblueprint/girls-write-now)

**Review by kylezryr — CHANGES_REQUESTED** [review]
everything looks great! i cant seem to test out the functionality of logging in/signing up because i think im missing some env variables but i trust it works 🙏 
just a change to make regarding Expo Router stack definitions to get cleaner pathnames for routing and following conventions! great work stephen 🫡

File: src/app/_layout.tsx (hunk 11 above)
Comment (rachaelch3n) [reply src/app/_layout.tsx:9]: i think this is my bad 😅 did not realize file structure for mobile was meant to look like this

File: src/app/auth/signup/page.tsx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+        password,
+        password_confirmation: password,
+        first_name,
+        last_name,
+        phone: phone_number,
+        // You'll need to add these fields to your form or provide defaults
+        zone_id: 1, // TODO: Add zone selection to form
+        home_lat: 0, // TODO: Add location picker to form
+        home_lon: 0, // TODO: Add location picker to form
+        onfleet_worker_id: '', // TODO: Add OnFleet ID to form or set after signup
```
Comment (rachaelch3n) [inline src/app/auth/signup/page.tsx]: anchit confirmed that we can just get rid of these fields. you should also remove them from rails too (run a migration)

Comment (stephenhungg) [reply src/app/auth/signup/page.tsx]: fixed on new branch
